### PR TITLE
Vectorize cleamfigure a bit

### PR DESCRIPTION
**Pull request by miscco**
Rewrite the coarsenLine function to take advantage of matrix operations.
	modified:   src/cleanfigure.m

### CODE REVIEW

**Comment by okomarov** [comment]
Hi @miscco,

We appreciate your contribution, however we cannot accept pulls against our master branch, but we do on our `develop` branch. (See the guidelines on https://github.com/matlab2tikz/matlab2tikz/wiki/Recommended-git-workflow.)

Also, we already included into the develop branch, a proper point-reduction algorithm, thus deprecating `coarsenLine()`.

**Comment by miscco** [comment]
Hey Oleg, sorry for the noise.,
I have to have a closer look on the new function. After a glance at the new code I would guess that the vectorization would still be useful, but I have to check it when I have some spare time.

**Comment by okomarov** [comment]
Be aware that we already tried to speed that up without major success. Also, we stress tested the current version on plots with millions of data points and it was still fast. I can't dig up the whole discussion for reference right now.

Sent from my iPhone

On 19 Aug 2015, at 11:15, "Michael Schellenberger" <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey Oleg, sorry for the noise.,
I have to have a closer look on the new function. After a glance at the new code I would guess that the vectorization would still be useful, but I have to check it when I have some spare time.

File: src/cleanfigure.m [hunk 1]
```diff
@@ -347,43 +348,49 @@ function coarsenLine(meta, handle, minimumPointsDistance)
...
-  % Make sure to keep NaNs.
-  mask = mask | any(isnan(data)')';
+  % To eliminate false positives for consecutive finds, recursively remove 
+  % the first close data points and iterate as long as there are some
+  while(~isempty(find(mask==0, 1))) 
 
+    % Compute the distance of between the points,
+	% incorporating possible log-scaling of the axes.
+	visDiff = [0.0, 0.0;diff(data)];  
+	if isXlog
```
Comment (PeterPablo) [inline src/cleanfigure.m]: wrong whitespace

**Comment by PeterPablo** [comment]
Thank you for your submission!

The code applies to `develop` as well. Please resubmit and post some speed comparisons (e.g. `tic`, `toc`). Travis continious integration tests failed so your solution is different to what currently is implemented. As such, please check the results of the testsuite to verify, if your changes do not cause regressions.

**Comment by PeterPablo** [comment]
Continues in #736

**Pull request closed without merging**